providers/asrockrack/firmware.go

```diff
@@ -63,16 +63,31 @@ func (a *ASRockRack) FirmwareInstallStatus(ctx context.Context, installVersion,
 func (a *ASRockRack) firmwareInstallBMC(ctx context.Context, reader io.Reader, fileSize int64) error {
 	var err error
 
+	// 0. take the model so that we use a different endpoint on E3C256D4ID-NL
+	device := common.NewDevice()
+	device.Metadata = map[string]string{}
+	err = a.fruAttributes(ctx, &device)
+	if err != nil {
+		a.log.V(2).WithValues("step", "0/4").Info("failed to get model, err: " + err.Error())
+		a.log.V(2).WithValues("step", "0/4").Info("continue with default endpoint")
```

Comment:
I suggest this returns an error here - if the model cannot be identified, its unlikely to be supported

Reply:
Until now, we used the default endpoint for every model, and now only E3C256D4I needs a special one.
So, if the model is unknown, we still try the default one.

However, the code you mention test the error of `fruAttributes` and it's true that if we fail on getting them, we might have a real problem.